src/main/java/com/icthh/xm/ms/configuration/config/BeanConfiguration.java

```diff
@@ -99,4 +140,73 @@ public void onRefresh(String config) {
         };
     }
 
+    @Bean
+    @ConditionalOnProperty(prefix = "application.config-repository.s3", name = "enabled", havingValue = "true")
+    public S3Client s3Client(ApplicationProperties applicationProperties) {
+        var s3Config = applicationProperties.getConfigRepository().getS3();
+        var credentials = AwsBasicCredentials.create(s3Config.getAccessKey(), s3Config.getSecretKey());
+        return S3Client.builder()
+                .endpointOverride(URI.create(s3Config.getEndpoint()))
+                .credentialsProvider(StaticCredentialsProvider.create(credentials))
+                .region(Region.of(s3Config.getRegion()))
+                .serviceConfiguration(buildS3Configuration(s3Config))
+                .build();
+    }
+
+    private static S3Configuration buildS3Configuration(S3 s3Config) {
+        return S3Configuration.builder()
+                .pathStyleAccessEnabled(s3Config.getPathStyleAccess())
+                .build();
+    }
+
+    private static S3Repository createS3Repository(S3 s3, S3Client s3Client) {
+        log.info("Creating s3 repository");
+        return new S3Repository(s3Client, s3.getBucket(), s3.getConfigPath());
+    }
+
+    private static JGitRepository createJGitRepository(ApplicationProperties applicationProperties, Lock lock,
```

Comment:
why you create jgit manually, just put it to method params, and spring will create it

Reply:
I did this to always have one Bean of PersistenceConfigRepository("configRepository") in application context. I can try to refactor to ConditionalOnProperty way with different beans, but in the case of a dynamic repository we still have 3 beans(4 - without removing "configRepository" bean) and we need to qualify which ones we want to use in our logic. For case with git or s3 mode  - 2 same beans if we still use "configRepository" or 1 if we delete that.